app/api/speakers.py

```diff
@@ -0,0 +1,155 @@
+    mobile = fields.Str(allow_none=True)
+    website = fields.Url(allow_none=True)
+    twitter = fields.Url(allow_none=True)
+    facebook = fields.Url(allow_none=True)
+    github = fields.Url(allow_none=True)
+    linkedin = fields.Url(allow_none=True)
+    organisation = fields.Str(allow_none=True)
+    is_featured = fields.Boolean(default=False)
+    position = fields.Str(allow_none=True)
+    country = fields.Str(allow_none=True)
+    city = fields.Str(allow_none=True)
+    gender = fields.Str(allow_none=True)
+    heard_from = fields.Str(allow_none=True)
+    sponsorship_required = fields.Str(allow_none=True)
+    event = Relationship(attribute='event',
+                         self_view='v1.speaker_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'speaker_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+    user = Relationship(attribute='user',
+                        self_view='v1.speaker_user',
+                        self_view_kwargs={'id': '<id>'},
+                        related_view='v1.user_detail',
+                        related_view_kwargs={'speaker_id': '<id>'},
+                        schema='UserSchema',
+                        type_='user')
+    sessions = Relationship(attribute='sessions',
+                            self_view='v1.speaker_session',
+                            self_view_kwargs={'id': '<id>'},
+                            related_view='v1.session_list',
+                            related_view_kwargs={'speaker_id': '<id>'},
+                            schema='SessionSchema',
+                            type_='session')
+
+
+class SpeakerList(ResourceList):
+    """
+    List and create speakers
+    """
+    def query(self, view_kwargs):
+        """
+        query method for speakers list class
+        :param view_kwargs:
+        :return:
+        """
+        query_ = self.session.query(Speaker)
+        if view_kwargs.get('event_id'):
+            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
+            query_ = query_.join(Event).filter(Event.id == event.id)
+        elif view_kwargs.get('event_identifier'):
+            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
+            query_ = query_.join(Event).filter(Event.id == event.id)
+        if view_kwargs.get('user_id'):
+            user = safe_query(self, User, 'id', view_kwargs['user_id'], 'user_id')
+            query_ = query_.join(User).filter(User.id == user.id)
+        if view_kwargs.get('session_id'):
+            session = safe_query(self, Session, 'id', view_kwargs['session_id'], 'session_id')
+            # session-speaker :: many-to-many relationship
+            query_ = Speaker.query.filter(Speaker.sessions.any(id=session.id))
```

Comment:
@shubham-padia That won't work because in the case that there are no sessions with the session_id passes, it will return an empty list, instead of giving an error. What I have written using safe query first checks if there is actually a session with the given id, then proceeds to fetch related speakers.